pyteal/ast/abi/address.py

```diff
@@ -93,6 +93,7 @@ def set(
 
         match value:
             case ComputedValue():
+                # TODO need to see again if we need to change on type system or not, _set_with_computed_type?
                 pts = value.produced_type_spec()
                 if pts == AddressTypeSpec() or pts == StaticArrayTypeSpec(
                     ByteTypeSpec(), AddressLength.Bytes
```

Comment:
question on this part: we let `type_spec_is_assignable_to` to be false when we want to check if `byte[32]` is assignable to `address`, while in this case, we do want to assign `byte[32]` to `address`.

What should be changed:
- the type system?
- or the implementation, say we leave a note here for special corner case?

Reply:
It's a good question.  

My 2c and happy to discuss live: I favor changing the type system because it feels analogous to the bi-directional treatment of `NameTuple` and `Tuple`.

Reply:
I agree the type system to be changed to not allow `Address().set(StaticArrary[Bytes, 32]())` (and the `ComputedValue` variant)

Reply:
change reflected in [3e776f8](https://github.com/algorand/pyteal/pull/567/commits/3e776f8a2bdf00422fe5dbab865450d7b46ac906)

Reply:
defer `unsafe_cast` implementation to a PR after this one get merged.

Reply:
@ahangsu Thanks for the changes + `unsafe_cast` in follow on PR - I'm good to resolve. 

I have a new open question, but can move discussion out of this thread on my side.

Reply:
just a reminder, #580 carries the `unsafe_cast` implementation, we can always rebase to this PR and get all the code together.

The final question is on another thread, i.e., on the type signature of `Address.set`, and the behavior related with mypy.